### PR TITLE
fix(hcl2cdk): flaky tests on hcl2cdk expressions

### PR DESCRIPTION
**Pull request by mutahhir**
### Related issue

Fixes CI issues on release

### Description

As per the following article on jest, we need to return all `resolves` or `rejects` expects, and that's what caused the test to be flaky. The test itself was failing, which was correct, but only failed when it accidentally ran the full thing:
image

### Checklist

- [ ] I have updated the PR title to match [CDKTF's style guide](https://github.com/hashicorp/terraform-cdk/blob/main/CONTRIBUTING.md#pull-requests-1)
- [ ] I have run the linter on my code locally
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation if applicable
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works if applicable
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/@cdktf/hcl2cdk/test/expressionToTs.test.ts [hunk 1]
```diff
@@ -253,20 +253,6 @@ describe("expressionToTs", () => {
...
-    const result = await convertTerraformExpressionToTs(
-      expression,
-      scope,
-      getType
-    );
-    expect(code(result)).toMatchInlineSnapshot(
-      `"\\"\${\\" + awsS3BucketExamplebucket.networkInterface + \\"}[0].access_config[0].assigned_nat_ip\\""`
-    );
-  });
-
```
Comment (ansgarm) [inline packages/@cdktf/hcl2cdk/test/expressionToTs.test.ts:269]: Why is this removed?

**Review by ansgarm — APPROVED** [review]

**Review by DanielMSchmidt — APPROVED** [review]
